olp-cpp-sdk-dataservice-read/src/VersionedLayerClientImpl.cpp

```diff
@@ -324,7 +324,14 @@ client::CancellationToken VersionedLayerClientImpl::PrefetchTiles(
       // Because the handling of prefetch tiles responses is performed by the
       // inner-task, no need to set a callback here. Otherwise, the user would
       // be notified with empty results.
-      nullptr);
+      // It is possible to not invoke inner task, when it was cancelled before
+      // execution.
+      [callback](EmptyResponse response) {
```

Comment:
Be aware that this is not enough. You need to remove additionaly, if possible, all the user callback triggers from the above lambda and fwd the errors only to this lambda.

Reply:
Also please use OLPEDGE-1005 as ticket.

Reply:
Added error propagation from task.